### ISSUE TITLE
Switch from Bitmap.recycle() to more modern APIs

### ISSUE THREAD

**Issue by Blaisorblade** [farkam135/GoIV#440]
We only support Android >= 4.4, and Bitmap.recycle() isn't needed there.

But for those versions docs have different performance tips we should take a look at:

https://developer.android.com/training/displaying-bitmaps/manage-memory.html#inBitmap

*Maybe* we can just remove all `recycle()` calls and think separately about further optimizations, but if the tips mean "replace calls to `recycle()` by something else in the same locations" we should do the change at once.

**Comment by Blaisorblade** [comment]
We might also want to review suggestions on https://developer.android.com/training/displaying-bitmaps/index.html. In particular, Android complains *CONSTANTLY* that we do too much work on the main thread, and that suggests that bitmap processing should happen on other threads.

**Comment by Blaisorblade** [comment]
Yep, often, about (IIRC) ~50-100 dropped frames for GoIV, on a Nexus 5 (possibly overheated). If you don't see that, your phone is probably faster or has better software ^_^. But we totally violate guidelines.

**Comment by sarav** [comment]
I didn't check the logs. But I know why it's complaining and I don't think we should care to fix it. Those complaints matter if we have an active UI when those "too much work on main thread" happens. When we are processing bitmaps, we have no UI to show. So, let's not complicate our code by trying to address a false warning.

The main reason for that complaint is that if you do too much work in the background in the main thread and you have some UI shown to the use, their interactions can be laggy. When we show UI, there's no background work going on.

**Comment by sarav** [comment]
As discussed over chat, most services run in the same thread as the UI. So, for example, if an app like Gmail used a service to download emails but didn't create a separate thread for the service, then the UI would freeze and lag like crazy since downloads can take seconds to finish. Bitmap processing and OCR is also a lot of work to do in the main thread _**IF**_ we had an active UI in that time period. Luckily, we don't.

**Comment by Blaisorblade** [comment]
FWIW: there is at least *one* UI-sensitive computation, the IV computation. I'm not sure a loop over 4096 possibilities is that expensive, but any warning there might be worth checking. (Not that GoIV actually freezes much, so maybe that's also irrelevant).
